### PR TITLE
Implement "introduce ship" in relay compontent

### PR DESCRIPTION
**Pull request by bfwg**
Hi, 
I implemented the mutation on app side, now the example not only can add ship from `POST` requests but also from the app its self.

### CODE REVIEW

File: examples/star-wars/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "graphql-relay": "^0.3.3",
     "react": "^0.14.0",
     "react-dom": "^0.14.0",
-    "react-relay": "file:../../",
+    "react-relay": "^0.4.0",
```
Comment (josephsavona) [inline examples/star-wars/package.json]: why this change?

**Comment by bfwg** [comment]
@jabagawee 
Oh I forgot to change it back to "react-relay": "file:../../",

**Comment by jabagawee** [comment]
me? i think you meant to mention someone else here

**Comment by bfwg** [comment]
@josephsavona 
Oh I forgot to change it back to "react-relay": "file:../../",
@jabagawee 
Sorry, my eyes isn't working today :P

**Comment by josephsavona** [comment]
@kassens or @steveluscher can you take a look?

File: examples/star-wars/data/database.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+"use strict";
```
Comment (steveluscher) [inline examples/star-wars/data/database.js]: Can you throw these below the license header as in the rest of the files in `src/`

File: examples/star-wars/data/schema.js [hunk 3]
```diff
@@ -26,12 +27,14 @@ import {
   fromGlobalId,
   globalIdField,
   mutationWithClientMutationId,
+  cursorForObjectInConnection,
```
Comment (steveluscher) [inline examples/star-wars/data/schema.js]: Nit: alpha order imports.

File: examples/star-wars/data/schema.js [hunk 4]
```diff
@@ -26,12 +27,14 @@ import {
...
   nodeDefinitions,
 } from 'graphql-relay';
 
 import {
   getFaction,
   getShip,
+  getShips,
```
Comment (steveluscher) [inline examples/star-wars/data/schema.js]: Hunh. These weren't alpha ordered. Mind fixing them up?

File: examples/star-wars/data/schema.js [hunk 5]
```diff
@@ -171,8 +174,9 @@ var shipType = new GraphQLObjectType({
  *     node: Ship
  *   }
  */
-var {connectionType: shipConnection} =
-  connectionDefinitions({name: 'Ship', nodeType: shipType});
+var {connectionType: shipConnection,
```
Comment (steveluscher) [inline examples/star-wars/data/schema.js]: Nit: style.

```
var {
  every: variable,
  on: a,
  different: line,
} = …;
```

File: examples/star-wars/data/schema.js [hunk 6]
```diff
@@ -189,6 +193,11 @@ var factionType = new GraphQLObjectType({
   description: 'A faction in the Star Wars saga',
   fields: () => ({
     id: globalIdField('Faction'),
+    fid: {
+      type: GraphQLString,
+      description: 'id of faction in db',
+      resolve: (faction) => faction.id
```
Comment (steveluscher) [inline examples/star-wars/data/schema.js]: Add a trailing comma here.

File: examples/star-wars/data/schema.js [hunk 7]
```diff
@@ -259,9 +268,15 @@ var shipMutation = mutationWithClientMutationId({
...
-      resolve: (payload) => getShip(payload.shipId)
+    newShipEdge: {
+      type: ShipEdge,
+      resolve: (payload) => {
+        var ship = getShip(payload.shipId);
+        return {
+          cursor: cursorForObjectInConnection(getShips(payload.factionId), ship),
+          node: ship,
+        };
+      }
```
Comment (steveluscher) [inline examples/star-wars/data/schema.js]: Add a trailing comma here.

File: examples/star-wars/js/components/StarWarsApp.js [hunk 8]
```diff
@@ -13,23 +13,72 @@
...
 import Relay from 'react-relay';
 import StarWarsShip from './StarWarsShip';
+import AddShipMutation from '../mutation/AddShipMutation';
 
 class StarWarsApp extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      shipName: '',
+      factionId: 1
```
Comment (steveluscher) [inline examples/star-wars/js/components/StarWarsApp.js]: Add a trailing comma here and alpha order these properties.

File: examples/star-wars/js/components/StarWarsApp.js [hunk 9]
```diff
@@ -13,23 +13,72 @@
...
+    };
+  }
+
+  addship() {
```
Comment (steveluscher) [inline examples/star-wars/js/components/StarWarsApp.js]: This is an event handler. Can we name it `handleAddShipClick` to make this clear? Alpha order it with the other handle-type methods.

File: examples/star-wars/js/components/StarWarsApp.js [hunk 10]
```diff
@@ -13,23 +13,72 @@
...
+              <ol>
+                {faction.ships.edges.map(edge => (
+                  <li><StarWarsShip ship={edge.node} /></li>
+                ))}
+              </ol>
+            </li>
+          ))}
+            <li>
+              <h1>Introduce Ship</h1>
+              <ol>
```
Comment (steveluscher) [inline examples/star-wars/js/components/StarWarsApp.js]: Rather than change tracking the text input, what do you think about a simpler design where this is made a bonafide `<form>` with an `onSubmit` handler. In your `handleAddShipSubmit` method, you can simply pull the ship name and faction from refs, and then reset the form.

```js
handleAddShipSubmit(e) {
  e.preventDefault();
  Relay.Store.update(
    new AddShipMutation({
      faction: this.props.factions[this.refs.factionIdSelect.value],
      name: this.refs.shipNameInput.value,
    })
  );
  this.refs.shipNameInput.value = '';
}
```

```js
<form onSubmit={this.handleAddShipSubmit.bind(this)}>
  <input type="text" ref="shipNameInput" />
  <select ref="shipNameInput" />…</select>
</form>
```

File: examples/star-wars/js/components/StarWarsApp.js [hunk 11]
```diff
@@ -38,6 +87,8 @@ export default Relay.createContainer(StarWarsApp, {
   fragments: {
     factions: () => Relay.QL`
       fragment on Faction @relay(plural: true) {
+        id,
+        fid,
```
Comment (steveluscher) [inline examples/star-wars/js/components/StarWarsApp.js]: Calling this `factionId` will make it more apparent what's being fetched here.

**Comment by steveluscher** [comment]
Sorry for the delay. This is a great start!

I've inlined some comments. You'll have to rebase after you take care of them. Also, do a quick once over of the code for missing trailing commas and alpha ordered properties in object literals. I'll do another round of feedback with you when you're done!

**Comment by yungsters** [comment]
I've rebased this and addressed all feedback in #892.
